src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -292,6 +304,7 @@ ucs_status_t uct_rdmacm_cm_ep_disconnect(uct_ep_h ep, unsigned flags)
         return UCS_ERR_IO_ERROR;
     }
 
+    cep->flags |= UCT_RDMACM_CM_EP_DISCONNECT_CALL;
```

Comment:
should not it be before line 299  `if (rdma_disconnect(cep->id)) {` ?

Reply:
if rdma_disconnect wasn't successful, shouldn't we allow the user to try again?

Reply:
I don't think that further calls might be successful in this case, need to set EP failed and user should just destroy it